### DISCUSSION TITLE
Prevent Admins from viewing uploaded user files

### DISCUSSION THREAD

**Discussion by tkg61** [open-webui/open-webui#5023] [General]
**Is your feature request related to a problem? Please describe.**
In this [commit](https://github.com/open-webui/open-webui/commit/4519ddd0e97b33114d5ad0d012ed4523937f55c7) it was fixed so that files would remain private to their own users but it also seems like admins still can view all files (openwebui.url/api/v1/files). 

i would like to request the same feature as [preventing the admins from seeing user chats](https://github.com/open-webui/open-webui/discussions/2807) and have admins also not see user files.

**Describe the solution you'd like**
a configuration item that prevents Admins from seeing user uploaded files for privacy and insider threat reasons

e.g. ENABLE_ADMIN_FILE_ACCESS=false

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by silentoplayz** [comment]
> I would like to request the same feature as [preventing the admins from seeing user chats](https://github.com/open-webui/open-webui/discussions/2807)

This has already been implemented as an environment variable `ENABLE_ADMIN_CHAT_ACCESS=false`. [Here](https://github.com/open-webui/open-webui/pull/4351) is the related merged PR, for reference.

As for Admin accounts still being able to view all files uploaded on the `/api/v1/files` endpoint, I'm not sure if that is considered a bug or if it is an intended feature. I'll leave this issue open for hopefully another contributor to come along and provide you with a better answer.

**Comment by tkg61** [comment]
I was hoping it would be viewed as a feature request to complement what work has been done in what you linked.

**Comment by Hugobox** [comment]
Hi! I've added the flag ENABLE_ADMIN_CHAT_ACCESS=false on openwebui 0.3.23, and I see that the admin panel does not have access to user chats via the UI anymore, as expected, and the access to other users chats is also restricted through the API. 

But the access to files is still enabled via the API (for admins, users are restricted as expected), we really need a ENABLE_ADMIN_FILE_ACCESS env var as well.

1. Using the API, GET all files: /api/v1/docs#/files/list_files_files__get
2. Notice that you see all filenames of all files.
3. Copy the Id of any file.
4. GET file content by Id: /api/v1/docs#/files/get_file_content_by_id_files__id__content_get
5. Paste Id and get file contents

How can we make sure files remain accessible only by their respective owners?
